### PR TITLE
Program to rotate array of size n by d elements. ISSUE NO. #259

### PR DESCRIPTION
**Pull request by rishabhgarg25699**

### CODE REVIEW

**Comment by rishabhgarg25699** [comment]
No,It take more than 1 sec. Because time complexity of this program is O(n^rotations)
According to your question, n=100 and rotation =10^7.So it take 10^9 operations means it take more than 1 sec.
Yes we can reduce the time complexity of this program.

File: Algorithms/Computational-Geometry/arrayRotation.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
+//Name = Rishabh Garg
```
Comment (arnav-mandal1234) [inline Algorithms/Computational-Geometry/arrayRotation.c]: improve the time complexity.
Writing brute algorithms won't suffice.

File: Algorithms/Computational-Geometry/Array program.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                if(small>(a[i]+a[j]))
+                {
+                    small=a[i]+a[j];
+                }
+            }
+        }
+    //After running both loops we have a smallest addition of two variables 
+    //in small variable.Now we print this value.
+        printf("%d\n",small);
+    return 0;
```
Comment (arnav-mandal1234) [inline Algorithms/Computational-Geometry/Array program.cpp]: indentation problem in many areas.

Comment (rishabhgarg25699) [reply Algorithms/Computational-Geometry/Array program.cpp]: Its my old PR . Now it is closed. Please review my new pull request @arnav-mandal1234

**Comment by rishabhgarg25699** [comment]
Now please review my code . I have tried my best to explain the logic very clearly .

**Comment by rishabhgarg25699** [comment]
@ramitsawhney27  PLEASE REVIEW MY CODE ASAP.

File: Algorithms/Computational-Geometry/arrayRotation.c [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	printf("Enter how much do you want to left rotate the array\n ");
+	scanf("%d",&rotations);
+	//The below function has the main logic of the program.In this we have to find GCD.
+	//The purpose of finding of GCD is that when we got the n and rotations, then we calculate the 
+	//GCD of the both the numbers.Then that array will broke into groups of size no. equal to GCD.
+	//This GCD helps to minimize the comparison within the array.The temination condition came when
+	//GCD goes down to 1.If GCD is one then elements will be moved within their small patch.
+	//With the help of GCD, elements jumps directily leaving rotation variable in one loop.
+	//By this way also,this method decreases the time complexity also. 
+	for (i = 0; i < gcd(rotations, n); i++)
```
Comment (sahilbansal17) [inline Algorithms/Computational-Geometry/arrayRotation.c]: This part is directly copied from [geeksforgeeks](https://www.geeksforgeeks.org/array-rotation/). Closing the PR.

Comment (rishabhgarg25699) [reply Algorithms/Computational-Geometry/arrayRotation.c]: No,I did not copied this code from anywhere

**Pull request closed without merging**